user-interface/eslint-ui-test.config.mjs

```diff
@@ -0,0 +1,44 @@
+import eslintUiConfig from './eslint-ui.config.mjs';
+import { createRequire } from 'module';
+
+const require = createRequire(import.meta.url);
+
+const tsEslint = require('typescript-eslint');
+const testingLibrary = require('eslint-plugin-testing-library');
+
+/**
+ * eslintUiTestConfig
+ *
+ * This ConfigArray is intended to be the eslint configuration for TypeScript
+ * family test files (e.g. `.test.ts`, `.test.tsx`).
+ */
+const eslintUiTestConfig = tsEslint.config(eslintUiConfig, {
+  plugins: testingLibrary.configs['flat/react']['plugins'],
+  files: ['**/*.test.ts', '**/*.test.tsx'],
+  rules: {
+    'testing-library/await-async-events': 'error',
+    'testing-library/await-async-queries': 'error',
+    'testing-library/await-async-utils': 'warn', // default: error
+    'testing-library/no-await-sync-events': 'error',
+    'testing-library/no-await-sync-queries': 'error',
+    'testing-library/no-container': 'warn', // default: error
+    'testing-library/no-debugging-utils': 'warn',
+    'testing-library/no-dom-import': 'error',
+    'testing-library/no-global-regexp-flag-in-query': 'error',
+    'testing-library/no-manual-cleanup': 'error',
+    'testing-library/no-node-access': 'off', // default: error
+    'testing-library/no-promise-in-fire-event': 'error',
+    'testing-library/no-render-in-lifecycle': 'warn', // default: error
+    'testing-library/no-unnecessary-act': 'error',
+    'testing-library/no-wait-for-multiple-assertions': 'warn', // default: error
+    'testing-library/no-wait-for-side-effects': 'warn', // default: error
+    'testing-library/no-wait-for-snapshot': 'error',
+    'testing-library/prefer-find-by': 'error',
+    'testing-library/prefer-presence-queries': 'warn', // default: error
+    'testing-library/prefer-query-by-disappearance': 'error',
+    'testing-library/prefer-screen-queries': 'warn', // default: error
+    'testing-library/render-result-naming-convention': 'error',
+  },
```

Comment:
Using the provided flat config causes a lot of errors I'm not sure we care about. For example, `no-node-access` produces a lot of errors for DOM traversal which I'm not sure we want to follow their rule. There are others that may be good to move toward, but would cause major refactoring of some of our tests such as `no-wait-for-multiple-assertions`. I've marked all rules that deviate from the default with a comment.

Reply:
If we decide we want to not turn on a rule we could leave it here and set to `off`, or we could remove it completely here.